File: Binary_Heap/min_binary_heap_insert.c
/* File: min_binary_heap_insert.c */

#include <stdio.h>

#define MAX 100

int heap[MAX];
int size = 0;

/* Swap function */
void swap(int *a, int *b)
{
    int temp = *a;
    *a = *b;
    *b = temp;
}

/* Insert into Min Heap */
void insert(int value)
{
    size++;
    int i = size;

    heap[i] = value;

    /* Heapify Up */
    while (i > 1 && heap[i] < heap[i/2])
    {
        swap(&heap[i], &heap[i/2]);
        i = i/2;
    }
}

/* Print Heap */
void printHeap()
{
    for(int i = 1; i <= size; i++)
        printf("%d ", heap[i]);

    printf("\n");
}

int main()
{
    insert(20);
    insert(15);
    insert(30);
    insert(5);
    insert(10);

    printf("Min Heap Elements:\n");
    printHeap();

    return 0;
}

File: Binary_Heap/MinBinaryHeapInsert.java
// File: MinBinaryHeapInsert.java

class MinBinaryHeapInsert {

    int[] heap = new int[100];
    int size = 0;

    void insert(int value)
    {
        size++;
        int i = size;

        heap[i] = value;

        while(i > 1 && heap[i] < heap[i/2])
        {
            int temp = heap[i];
            heap[i] = heap[i/2];
            heap[i/2] = temp;

            i = i/2;
        }
    }

    void printHeap()
    {
        for(int i = 1; i <= size; i++)
            System.out.print(heap[i] + " ");

        System.out.println();
    }

    public static void main(String[] args)
    {
        MinBinaryHeapInsert h = new MinBinaryHeapInsert();

        h.insert(20);
        h.insert(15);
        h.insert(30);
        h.insert(5);
        h.insert(10);

        System.out.println("Min Heap Elements:");
        h.printHeap();
    }
}
